Move ${STEP} location in analyses prep(dev) and stats/plots (ops)

AndrewBenjamin-NOAA opened the pull request:
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

CO has requested that EVS script/job names match the EVS com/ structure: …/com/evs/v1.0/${STEP}/${COMPONENT}/…
This PR updates analyses prep that is run only in dev, and analyses stats and plots script names by moving ${STEP} earlier in the filename. Tests showed that all updates worked for prep and stats, with no errors or warnings.

## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by?
No
* Do you have any planned upcoming annual leave/PTO?
No
* Are there any changes needed in the times when the jobs are supposed to run/kick-off?
No
  
- [x] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [x] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [x] References the feature branch for `HOMEevs` are removed from the code.
- [x] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [x] Jobs over 15 minutes in runtime have restart capability.
- [x] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [x] Jobs contain the appropriate file checking and don't run METplus for any missing data.
- [x] Code is using METplus wrappers structure and not calling MET executables directly.
- [x] Log is free of any ERRORs or WARNINGs.

## Testing Instructions
git clone https://github.com/AndrewBenjamin-NOAA/EVS.git
cd EVS/
git checkout feature/analyses_step_update
ln -sf /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix fix
cd sorc/
./build.sh

Please edit and test the analyses dev-only prep and stats jobs, as well as the stats  and plots dev drivers for ops EVS.

**dev_only**
Things to change in the dev_only drivers:
- HOMEevs to the directory that you are currently in (e.g., add /pr###test)
- Change COMIN to  COMIN=/lfs/h2/emc/vpppg/noscrub/**emc.vpppg**/**${NET}_devonly**/${evs_ver_2d}   (for stats)
- KEEPDATA=YES
- SENDMAIL=NO
- set VDATE=20251013

dev/drivers/scripts/prep/analyses/jevs_analyses_precip_prep.sh   (Run 4x: vhr= 00,06,12,18)

dev/drivers/scripts/stats/analyses/
*Run every hour (vhr=00-22), then once all those jobs complete, run vhr=23 for the following jobs:
- jevs_stats_analyses_ccpa_precip.sh
- jevs_stats_analyses_rtma_precip.sh
- jevs_stats_analyses_urma_precip.sh

**ops**
Things to change in the drivers:
- HOMEevs to the directory that you are currently in (e.g., add /pr###test)
- Change $USER to emc.vpppg in COMIN
- KEEPDATA=YES
- SENDMAIL=NO

dev/drivers/scripts/stats/analyses/
*Run every hour (vhr=00-22), then once all those jobs complete, run vhr=23 for the following jobs:
- jevs_stats_analyses_rtma_grid2obs.sh
- jevs_stats_analyses_rtma_ru_grid2obs.sh
- jevs_stats_analyses_urma_grid2obs.sh

dev/drivers/scripts/plots/analyses/
- jevs_plots_analyses_grid2obs_last31days.sh

To compare results, the emc.vpppg parallel output can be found here:
stats: /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v2.0/stats/analyses/

The emc.vpppg dev_only output can be found here:
prep: /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs_devonly/v2.0/prep/analyses
stats: /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs_devonly/v2.0/stats/analyses

AndrewBenjamin-NOAA committed:
moved  in analyses prep step

AndrewBenjamin-NOAA committed:
moved  for analysis stats

AndrewBenjamin-NOAA committed:
moved  in analyses plots

AndrewBenjamin-NOAA commented:
@AliciaBentley-NOAA 

I believe I also need to updated the .ecf scripts in ~/ecf/scripts/prep/analyses, ~/ecf/scripts/stats/analyses , and ~/ecf/scripts/plots/analyses.

I can make those changes and commit them here. My question is if I should also change the file names, and do the ecf script  names appear anywhere else that I would have to change?

AliciaBentley-NOAA commented:
@AndrewBenjamin-NOAA Can you please update the name of the analyses prep job dev driver? The contents of the file were changed, but it appears that the name was not yet. Thanks!

AndrewBenjamin-NOAA committed:
renamed and updated .ecf scripts

AndrewBenjamin-NOAA committed:
renamed analysis prep dev driver

AndrewBenjamin-NOAA commented:
> @AndrewBenjamin-NOAA Can you please update the name of the analyses prep job dev driver? The contents of the file were changed, but it appears that the name was not yet. Thanks!

Absolutely. I also updated and renamed the corresponding .ecf scripts. This PR should now be ready to move on to testing as time allows

AliciaBentley-NOAA commented:
@AndrewBenjamin-NOAA Thank you! I pulled in your changes and started the prep job for the four vhrs! You can find the log files at the paths below. Please let me know if I should rerun these on INITDATE=20251016. CC @MarcelCaron-NOAA 

### jevs_prep_analyses_precip.sh

Logs: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr851test/EVS/dev/drivers/scripts/prep/analyses/
DATA: /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/
prep: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/prep/analyses/atmos.20251017/

AndrewBenjamin-NOAA commented:
@AliciaBentley-NOAA Thanks, your output looks good to me. In terms of VDATE, I'll defer to you and @MarcelCaron-NOAA on specifics. These jobs are run in dev-only, so I'm not sure what's best to compare with the parallel (aka dev-only).

MarcelCaron-NOAA commented:
> Please let me know if I should rerun these on INITDATE=20251016. 

@AliciaBentley-NOAA I'm not sure, but I think the runs are fine as-is.  I agree with @AndrewBenjamin-NOAA  that the logfiles are clean and output directories look normal compared to "evs_devonly":
```
/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs_devonly/v2.0/prep/analyses/atmos.20251017/ccpa/
```

AliciaBentley-NOAA commented:
@AndrewBenjamin-NOAA @MarcelCaron-NOAA Confirming that the analyses dev-only prep jobs look good! I will move on to the dev-only analyses precip stats. 

```
jevs_prep_analyses_precip.o222426116:### Job 222426116.cbqs01 - Exit status is 0
jevs_prep_analyses_precip.o222426116:    resources_used.walltime = 00:00:00
jevs_prep_analyses_precip.o222426116:    ctime = Mon Oct 20 12:38:35 2025
jevs_prep_analyses_precip.o222426116:    Error_Path = clogin03.cactus.wcoss2.ncep.noaa.gov:/lfs/h2/emc/vpppg/noscrub
jevs_prep_analyses_precip.o222426116:    mtime = Mon Oct 20 12:38:52 2025
jevs_prep_analyses_precip.o222426116:    Resource_List.walltime = 00:10:00

jevs_prep_analyses_precip.o222426117:### Job 222426117.cbqs01 - Exit status is 0
jevs_prep_analyses_precip.o222426117:    resources_used.walltime = 00:00:00
jevs_prep_analyses_precip.o222426117:    ctime = Mon Oct 20 12:38:39 2025
jevs_prep_analyses_precip.o222426117:    Error_Path = clogin03.cactus.wcoss2.ncep.noaa.gov:/lfs/h2/emc/vpppg/noscrub
jevs_prep_analyses_precip.o222426117:    mtime = Mon Oct 20 12:38:56 2025
jevs_prep_analyses_precip.o222426117:    Resource_List.walltime = 00:10:00

jevs_prep_analyses_precip.o222426118:### Job 222426118.cbqs01 - Exit status is 0
jevs_prep_analyses_precip.o222426118:    resources_used.walltime = 00:00:00
jevs_prep_analyses_precip.o222426118:    ctime = Mon Oct 20 12:38:43 2025
jevs_prep_analyses_precip.o222426118:    Error_Path = clogin03.cactus.wcoss2.ncep.noaa.gov:/lfs/h2/emc/vpppg/noscrub
jevs_prep_analyses_precip.o222426118:    mtime = Mon Oct 20 12:39:01 2025
jevs_prep_analyses_precip.o222426118:    Resource_List.walltime = 00:10:00

jevs_prep_analyses_precip.o222430230:### Job 222430230.cbqs01 - Exit status is 0
jevs_prep_analyses_precip.o222430230:    resources_used.walltime = 00:00:00
jevs_prep_analyses_precip.o222430230:    ctime = Mon Oct 20 13:32:07 2025
jevs_prep_analyses_precip.o222430230:    Error_Path = clogin03.cactus.wcoss2.ncep.noaa.gov:/lfs/h2/emc/vpppg/noscrub
jevs_prep_analyses_precip.o222430230:    mtime = Mon Oct 20 13:32:25 2025
jevs_prep_analyses_precip.o222430230:    Resource_List.walltime = 00:10:00
```

```
[alicia.bentley@clogin04 atmos.20251017] cd ccpa/
[alicia.bentley@clogin04 ccpa] ls
ccpa.t00z.01h.hrap.conus.gb2  ccpa.t04z.01h.hrap.conus.gb2  ccpa.t08z.01h.hrap.conus.gb2  ccpa.t12z.01h.hrap.conus.gb2  ccpa.t16z.01h.hrap.conus.gb2  ccpa.t20z.01h.hrap.conus.gb2
ccpa.t01z.01h.hrap.conus.gb2  ccpa.t05z.01h.hrap.conus.gb2  ccpa.t09z.01h.hrap.conus.gb2  ccpa.t13z.01h.hrap.conus.gb2  ccpa.t17z.01h.hrap.conus.gb2  ccpa.t21z.01h.hrap.conus.gb2
ccpa.t02z.01h.hrap.conus.gb2  ccpa.t06z.01h.hrap.conus.gb2  ccpa.t10z.01h.hrap.conus.gb2  ccpa.t14z.01h.hrap.conus.gb2  ccpa.t18z.01h.hrap.conus.gb2  ccpa.t22z.01h.hrap.conus.gb2
ccpa.t03z.01h.hrap.conus.gb2  ccpa.t07z.01h.hrap.conus.gb2  ccpa.t11z.01h.hrap.conus.gb2  ccpa.t15z.01h.hrap.conus.gb2  ccpa.t19z.01h.hrap.conus.gb2  ccpa.t23z.01h.hrap.conus.gb2
```

AndrewBenjamin-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/analyses_step_update

AliciaBentley-NOAA commented:
@AndrewBenjamin-NOAA @MarcelCaron-NOAA I have run the ccpa, rtma, and urma precip stats jobs for VDATE=20251016. I do not see any errors or warnings in the log files! The final .stat files were produced. Please confirm.

**jevs_stats_analyses_ccpa_precip.sh
jevs_stats_analyses_rtma_precip.sh
jevs_stats_analyses_urma_precip.sh**

Logs: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr851test/EVS/dev/drivers/scripts/stats/analyses/
DATA: /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/
prep: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/analyses/

AndrewBenjamin-NOAA commented:
@AliciaBentley-NOAA The output and final stats files look good to me.

MarcelCaron-NOAA commented:
@AliciaBentley-NOAA @AndrewBenjamin-NOAA Agreed, the stats logs and output look normal.

AliciaBentley-NOAA commented on dev/drivers/scripts/stats/analyses/jevs_stats_analyses_urma_grid2obs.sh:
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
  
```
@AndrewBenjamin-NOAA Can you please remove the extra line after #!/bin/bash in `stats/analyses/jevs_stats_analyses_urma_grid2obs.sh`‎? Thanks!

AndrewBenjamin-NOAA committed:
removed line from jevs_stats_analyses_urma_grid2obs.sh

AndrewBenjamin-NOAA committed:
(fixed) removed line from jevs_stats_analyses_urma_grid2obs.sh

AliciaBentley-NOAA commented:
Thanks!

AliciaBentley-NOAA commented:
@AndrewBenjamin-NOAA I have run the rtma, rtma_ru, and urma grid2obs stats jobs for VDATE=20251018. The vhr=23 rtma and urma jobs are still running. 

I do not see any errors or warnings in the log files for rtm_ru or any of the rtma or urma jobs that have completed thus far! Please confirm against stats from the emc.vpppg parallel when the last 2 jobs jobs finish. 

**emc.vpppg parallel:** /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v2.0/stats/analyses/

**jevs_stats_analyses_rtma_grid2obs.sh
jevs_stats_analyses_rtma_ru_grid2obs.sh
jevs_stats_analyses_urma_grid2obs.sh**

Logs: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr851test/EVS/dev/drivers/scripts/stats/analyses/
DATA: /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/
prep: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/analyses/

CC @MarcelCaron-NOAA

MarcelCaron-NOAA commented:
@AliciaBentley-NOAA Looks like they've all finished running.  Output files look normal and the jobs completed cleanly.
✔️ jevs_stats_analyses_rtma_grid2obs
✔️ jevs_stats_analyses_rtma_ru_grid2obs
✔️ jevs_stats_analyses_urma_grid2obs

AliciaBentley-NOAA commented:
@AndrewBenjamin-NOAA I have kicked off the analyses plots job for VDATE=20251018! It takes over an hour to run, so most folks will be checking the output tomorrow morning. Thanks!

### jevs_plots_analyses_grid2obs_last31days.sh 

**Logs:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr851test/EVS/dev/drivers/scripts/plots/analyses/
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/
**Plots:** /lfs/h2/emc/ptmp/alicia.bentley/evs/v2.0/plots/analyses/atmos.20251018/

CC @MarcelCaron-NOAA

AliciaBentley-NOAA commented:
@AndrewBenjamin-NOAA @MarcelCaron-NOAA I'm off to a meeting until 9am, but please let me know if the analyses plots job ran correctly yesterday whenever you get the chance! If it ran correctly, we can probably merge this PR this morning and I can continue to work on Andrew's aqm ${STEP} job this afternoon. Thanks!

AndrewBenjamin-NOAA commented:
@AliciaBentley-NOAA @MarcelCaron-NOAA 
The analyses plots job looks like it ran correctly to me. Logs look good, and the tarball is the same as the emc.vpppg parallel.

MarcelCaron-NOAA commented:
@AliciaBentley-NOAA @AndrewBenjamin-NOAA, Agreed!  The plots job looks good.

AliciaBentley-NOAA reviewed (approved):
I have reviewed the coding changes made in this PR and I think that they look good. One dev driver name was adjusted during PR testing and worked correctly. All jobs completed without errors or warnings when the correct INITDATE and VDATEs were used. I approve this PR to be merged. 

@AndrewBenjamin-NOAA Can you please confirm that your fork is synced?

CC @MarcelCaron-NOAA

MarcelCaron-NOAA reviewed (approved):
I've reviewed and approve the files changed in this PR 👍 

CC: @AliciaBentley-NOAA, @AndrewBenjamin-NOAA

AndrewBenjamin-NOAA commented:
@AliciaBentley-NOAA my fork is synced!

SamiraArdani-NOAA reviewed (approved):
@AndrewBenjamin-NOAA and @AliciaBentley-NOAA, I reviewed the changes in this PR and they look good. I can approve this PR to be merged into develop.
